### PR TITLE
Make port command line arguments be conditional on ifdefs

### PR DESCRIPTION
**Pull request by andy31415**
Implementation of usage of these ports is actually conditional in AppMain, so reporting we support arguments without them actually working is very confusing.

Changes:
  - conditionally include members/arguments/help 
  - enable ports in light app (I believe all apps should have this enabled, probably by default and unsure why we did not do this at the start, however for now at least light app is better)
  - update the `.securedCommissionerPort` logic - it seemed to have defaulted to 5542 however then a +10 was added in the code using it, effectively making the default 5552 without an obvious user way to override - user override always had 10 added to it which seemd strange.

Conditional logic is in https://github.com/project-chip/connectedhomeip/blob/master/examples/platform/linux/AppMain.cpp#L358

```cpp
// ...

    initParams.operationalServicePort        = CHIP_PORT;
    initParams.userDirectedCommissioningPort = CHIP_UDC_PORT;

#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE || CHIP_DEVICE_ENABLE_PORT_PARAMS
    // use a different service port to make testing possible with other sample devices running on same host
    initParams.operationalServicePort        = LinuxDeviceOptions::GetInstance().securedDevicePort;
    initParams.userDirectedCommissioningPort = LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort;
#endif // CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
// ...
```

### CODE REVIEW

File: examples/platform/linux/Options.h [hunk 1]
```diff
@@ -43,13 +43,17 @@ struct LinuxDeviceOptions
...
+    uint32_t spake2pIterations = 0; // When not provided (0), will default elsewhere
+    uint32_t mBleDevice        = 0;
+    bool mWiFi                 = false;
+    bool mThread               = false;
+#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE || CHIP_DEVICE_ENABLE_PORT_PARAMS
+    uint32_t securedDevicePort         = CHIP_PORT;
+    uint32_t unsecuredCommissionerPort = CHIP_UDC_PORT;
+#endif // CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
+#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
+    uint32_t securedCommissionerPort = CHIP_PORT + 10;
```
Comment (bzbarsky-apple) [inline examples/platform/linux/Options.h:55]: Why the change from CHIP_PORT+2 to CHIP_PORT+10 here?

Comment (andy31415) [reply examples/platform/linux/Options.h:55]: I think I was trying to look at port logic and https://github.com/project-chip/connectedhomeip/blob/master/examples/platform/linux/AppMain.cpp#L405 seems to say +10 ... but the logic seems odd in general ... that +10 is not overridable. Will look at this a bit more.

Comment (bzbarsky-apple) [reply examples/platform/linux/Options.h:55]: So that's adding another +10 on top of whatever value we have here (where we are _also_ adding a +2 or +10 or whatever)?

Comment (andy31415) [reply examples/platform/linux/Options.h:55]: I do not understand this at all ... :

```
❯ rg securedCommissionerPort
examples/platform/linux/Options.h
52:    uint32_t unsecuredCommissionerPort = CHIP_UDC_PORT;
55:    uint32_t securedCommissionerPort = CHIP_PORT + 10;

examples/platform/linux/Options.cpp
56:    kDeviceOption_UnsecuredCommissionerPort = 0x100b,
107:    { "unsecured-commissioner-port", kArgumentRequired, kDeviceOption_UnsecuredCommissionerPort },
397:    case kDeviceOption_UnsecuredCommissionerPort:
398:        LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort = static_cast<uint16_t>(atoi(aValue));
405:        LinuxDeviceOptions::GetInstance().securedCommissionerPort = static_cast<uint16_t>(atoi(aValue));

examples/platform/linux/AppMain.cpp
361:    initParams.userDirectedCommissioningPort = LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort;
405:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort + 10,
406:                                    LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort,

examples/platform/nxp/se05x/linux/AppMain.cpp
394:    initParams.userDirectedCommissioningPort = LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort;
434:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort + 10,
435:                                    LinuxDeviceOptions::GetInstance().unsecuredCommissionerPort) == CHIP_NO_ERROR)
```

so it looks like:
  - this thing has a default `CHIP_PORT + n` (used to be 2, I changed it to 10 probably wrongly)
  - it is NEVER user as is, it is always used as a `+10`
  
I believe the correct thing to do is actually set it to a default (it seems to be +12) and then actually use it. Will make the changes.

Comment (andy31415) [reply examples/platform/linux/Options.h]: For clarity, current master has:

```
> rg '\WsecuredCommissionerPort\W'
examples/platform/nxp/se05x/linux/AppMain.cpp
434:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort + 10,

examples/platform/linux/AppMain.cpp
405:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort + 10,

examples/platform/linux/Options.cpp
385:        LinuxDeviceOptions::GetInstance
```

I updated this to:
```
❯ rg '\WsecuredCommissionerPort\W'
examples/platform/linux/AppMain.cpp
405:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort,

examples/platform/linux/Options.h
55:    uint32_t securedCommissionerPort = CHIP_PORT + 12; // TODO: why + 12?

examples/platform/linux/Options.cpp
405:        LinuxDeviceOptions::GetInstance().securedCommissionerPort = static_cast<uint16_t>(atoi(aValue));

examples/platform/nxp/se05x/linux/AppMain.cpp
434:    VerifyOrReturn(InitCommissioner(LinuxDeviceOptions::GetInstance().securedCommissionerPort,
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by arkq — APPROVED** [review]
